### PR TITLE
Implement config_entries

### PR DESCRIPTION
**Pull request by gjohansson-ST**
Hi

I have added the option to use config_entries i.e. configure the integration also using the integration page directly in HA

**NOTE**: I took the liberty of changing the parameter account_id with username instead

I have also changed accordingly in readme + info docs

### CODE REVIEW

File: custom_components/adax/config_flow.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+"""Adds config flow for Mill integration."""
```
Comment (Danielhiversen) [inline custom_components/adax/config_flow.py]: Mill -> adax
```suggestion
"""Adds config flow for Adax integration."""
```

File: custom_components/adax/config_flow.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+DATA_SCHEMA = vol.Schema(
+    {
+    vol.Required(CONF_USERNAME): str,
+    vol.Required(CONF_PASSWORD): str}
+)
+
+
+class AdaxConfigFlow(config_entries.ConfigFlow, domain="adax"):
+    """Handle a config flow for Mill integration."""
```
Comment (Danielhiversen) [inline custom_components/adax/config_flow.py]: Mill -> adax
```suggestion
    """Handle a config flow for Adax integration."""
```

File: custom_components/adax/config_flow.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            return self.async_show_form(
+                step_id="user", data_schema=DATA_SCHEMA, errors={},
+            )
+
+        username = user_input[CONF_USERNAME].replace(" ", "")
+        password = user_input[CONF_PASSWORD].replace(" ", "")
+
+        """
+        TEST LOGIN
+        """
```
Comment (Danielhiversen) [inline custom_components/adax/config_flow.py]: Remove the string

File: custom_components/adax/config_flow.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        data = {'username': username,
+            'password': password,
+            'grant_type': 'password'}
+
+        ACCESS_TOKEN_URL = API_URL + '/auth/token'
+
+        async with aiohttp.ClientSession() as session:
+            async with session.post(ACCESS_TOKEN_URL, data=data) as response:
```
Comment (Danielhiversen) [inline custom_components/adax/config_flow.py]:
```suggestion
            async with session.post(API_URL + '/auth/token', data=data) as response:
```

**Review by Danielhiversen — CHANGES_REQUESTED** [review]
Thanks. 👍 

Why change from account_id to username?

It is called account id in the Adax app, so I think that makes the most sense. Username can also easily be confused by login username for Adax

File: custom_components/adax/config_flow.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        """
+        TEST LOGIN
+        """
+        data = {'username': username,
+            'password': password,
+            'grant_type': 'password'}
+
+        ACCESS_TOKEN_URL = API_URL + '/auth/token'
+
+        async with aiohttp.ClientSession() as session:
```
Comment (Danielhiversen) [inline custom_components/adax/config_flow.py]: Do not create a new session, but use 
`from homeassistant.helpers.aiohttp_client import async_get_clientsession
`

**Review by Danielhiversen — CHANGES_REQUESTED** [review]

**Comment by gjohansson-ST** [comment]
> Thanks. 👍 
> 
> 
> 
> Why change from account_id to username?
> 
> 
> 
> It is called account id in the Adax app, so I think that makes the most sense. Username can also easily be confused by login username for Adax

Ok. Not a big problem so I'll change it back together with the other small things to change from your review.

File: README.md [hunk 6]
```diff
@@ -11,15 +11,25 @@ Custom component for using [Adax](https://adax.no/en/) heaters in Home Assistant
...
+Enter your remote secret as password
+
+2.
 In configuration.yaml:
 
 ```
 climate:
   - platform: adax
-    account_id: "112395"  # replace with your account ID (see Adax WiFi app, Account Section)
+    username: "112395"  # replace with your account ID (see Adax WiFi app, Account Section)
```
Comment (Danielhiversen) [inline README.md]:
```suggestion
    account_id: "112395"  # replace with your account ID (see Adax WiFi app, Account Section)
```

File: custom_components/adax/config_flow.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+                data={
+                    "grant_type": "password",
+                    "username": username,
+                    "password": password,
+                },
+            )
+
+    if response.status != 200:
+            _LOGGER.info("Adax: Failed to login to retrieve token: %d", response.status)
+            raise CannotConnect
```
Comment (Danielhiversen) [inline custom_components/adax/config_flow.py]:
```suggestion
        _LOGGER.info("Adax: Failed to login to retrieve token: %d", response.status)
        raise CannotConnect
```

File: custom_components/adax/strings.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    "abort": {
+      "already_configured": "[%key:common::config_flow::abort::already_configured_account%]"
+    },
+    "error": {
+      "connection_error": "[%key:common::config_flow::error::connection_error%]"
+    },
+    "step": {
+      "user": {
+        "data": {
+          "account_id": "[%key:common::config_flow::data::account_id%]",
```
Comment (Danielhiversen) [inline custom_components/adax/strings.json]: I do not think this one exists?

File: custom_components/adax/config_flow.py [hunk 9]
```diff
@@ -0,0 +1,83 @@
+"""Adds config flow for Adax integration."""
+import logging
+
+import voluptuous as vol
+import aiohttp
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
```
Comment (Danielhiversen) [inline custom_components/adax/config_flow.py]:
```suggestion
from homeassistant.const import CONF_PASSWORD
```

File: custom_components/adax/config_flow.py [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            try:
+                username = user_input["account_id"].replace(" ", "")
```
Comment (Danielhiversen) [inline custom_components/adax/config_flow.py]:
```suggestion
                account_id = user_input["account_id"].replace(" ", "")
```

File: custom_components/adax/config_flow.py [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+    vol.Required("account_id"): str,
+    vol.Required(CONF_PASSWORD): str}
+)
+
+async def validate_input(hass: core.HomeAssistant, username, password):
```
Comment (Danielhiversen) [inline custom_components/adax/config_flow.py]:
```suggestion
async def validate_input(hass: core.HomeAssistant, account_id, password):
```

File: info.md [hunk 12]
```diff
@@ -2,6 +2,13 @@ Async integration for Adax heaters
 
 [Buy me a coffee :)](http://paypal.me/dahoiv)
 
+{% if version_installed.replace(".","") | int <= 22  %}
+## Change in platform
+
+For versions above 0.2.2 the setting of account_id has been replaced with username.
```
Comment (Danielhiversen) [inline info.md]:
```suggestion
```

**Review by Danielhiversen — CHANGES_REQUESTED** [review]

File: custom_components/adax/strings.json (hunk 8 above)
Comment (gjohansson-ST) [reply custom_components/adax/strings.json]: Correct

File: README.md (hunk 6 above)
Comment (gjohansson-ST) [reply README.md]: Changed back to account_id

**Comment by gjohansson-ST** [comment]
I think with these last updates it's now accurate. Or at least I hope I didn't forget anything :)

File: info.md [hunk 13]
```diff
@@ -19,13 +24,19 @@ Please be careful and do NOT install this on production systems. Also make sure
 - Change set temperature and turn on/off
 
 
+## Configuration (2 options)
+
+1.
+Go to integration page in HA, press + and search for Adax
+Enter your account id as username
```
Comment (Danielhiversen) [inline info.md]:
```suggestion
Enter your account id
```

File: info.md [hunk 14]
```diff
@@ -19,13 +24,19 @@ Please be careful and do NOT install this on production systems. Also make sure
...
 
-## Install
+2.
 In configuration.yaml:
 
 ```
 climate:
   - platform: adax
-    account_id: "112395"  # replace with your account ID (see Adax WiFi app, Account Section)
+    username: "112395"  # replace with your account ID (see Adax WiFi app, Account Section)
```
Comment (Danielhiversen) [inline info.md]:
```suggestion
    account_id: "112395"  # replace with your account ID (see Adax WiFi app, Account Section)
```

**Review by Danielhiversen — APPROVED** [review]
Thanks

**Pull request merged**